### PR TITLE
Reuse a browser to do screenshots

### PR DESCRIPTION
**Pull request by hakubo**
since tests wait for element to be visible before doing screenshot it should be safe
and it can speed up tests by not launching/closing a browser all the time

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@
 npm-debug.log*
 yarn-error.log
 testem.log
+.vscode
```
Comment (hakubo) [inline .gitignore]: for now I've added it to gitignore,
If you think it's worth it - I could commit some vscode config that is compatible with this repo

Comment (mydea) [reply .gitignore]: That's fine for now, thank you :)

**Comment by mydea** [comment]
It seems like this PR also includes the promise-async changes - is that right?
About the change itself, I think it is a good one, but I fear there might be places where issues could arise (e.g. when running tests in parallel, or possibly other cases). If we do this, I suggest we add a switch to disable this behavior (e.g. reuseChromeTab, which can be true by default). What do you think?

**Comment by hakubo** [comment]
yeah, I think you're right this can lead to some race condition.
Maybe instead of creating new chrome browser for a screenshot
we could just open new tab? that already should be lighter on resources

**Comment by mydea** [comment]
Yeah, that might be a good middle ground!

**Comment by hakubo** [comment]
@mydea done :)

**Comment by mydea** [comment]
Looks good, but one more question - is it OK if we never close the browser instance? Or does that lead to a memory leak?

**Comment by hakubo** [comment]
Ha! Good question. I think if parent process finishes chrome should close as well. I'll test it later today.

**Comment by mydea** [comment]
I think that's fine for now, I'll merge this - thank you! :)

**Pull request merged**